parsons/salesforce/salesforce.py

```diff
@@ -0,0 +1,191 @@
+from simple_salesforce import Salesforce as _Salesforce
+from parsons.utilities import check_env
+from parsons import Table
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class Salesforce:
+    """
+    `Args:`
+        username: str
+            The Salesforce username (usually an email address). Not required if
+            ``SALESFORCE_USERNAME`` env variable is passed.
+        password: str
+            The Salesforce password. Not required if ``SALESFORCE_PASSWORD`` env variable is
+            passed.
+        security_token: str
+            The Salesforce security token that can be acquired or reset in
+            Settings > My Personal Information > Reset My Security Token.
+            Not required if ``SALESFORCE_SECURITY_TOKEN`` env variable is passed.
+        test_environment: bool
+            If ``True`` the client will connect to a Saleforce sandbox instance. Not required if
+            ``SALESFORCE_DOMAIN`` env variable is passed.
+    `Returns:`
+        Salesforce class
+    """
+
+    def __init__(self, username=None, password=None, security_token=None, test_environment=False):
+
+        self.username = check_env.check('SALESFORCE_USERNAME', username)
+        self.password = check_env.check('SALESFORCE_PASSWORD', password)
+        self.security_token = check_env.check('SALESFORCE_SECURITY_TOKEN', security_token)
+
+        if test_environment:
+            self.domain = check_env.check('SALESFORCE_DOMAIN', 'test')
+        else:
+            self.domain = None
+
+        self._client = None
+
+    def query(self, soql):
+        """
+        `Args:`
+            soql: str
+                The desired query in Salesforce SOQL language (SQL with additional limitations).
+                For reference, see `Salesforce SOQL documentation<https://developer.salesforce.com/docs/atlas.en-us.soql_sosl.meta/soql_sosl/sforce_api_calls_soql.htm>`_.
+        `Returns:`
+            list of dicts with Salesforce data
+        """ # noqa: E501,E261
+
+        q = Table(self.client.query_all(soql))
+        logger.info(f'Found {q.num_rows} results')
+        return q
+
+    def insert(self, object, data_table):
```

Comment:
I wonder if in keeping with our overly explanatory naming conventions, should the method be called `insert_record`. If so, I think that the same would and should apply to the other methods.